cms/djangoapps/import_from_modulestore/helpers.py

```diff
@@ -0,0 +1,436 @@
+
+        Processes the staged content files for a component, creating the
+        necessary file content and associating it with the component version.
+        """
+        block_olx = etree.tostring(block_to_import).decode('utf-8')
+
+        for staged_content_file_data in staged_content_files:
+            original_filename = staged_content_file_data.filename
+            file_basename = os.path.basename(original_filename)
+            file_basename_no_ext, _ = os.path.splitext(file_basename)
+
+            # Skip files not referenced in the block
+            if file_basename not in block_olx and file_basename_no_ext not in block_olx:
+                log.info(f"Skipping file {original_filename} as it is not referenced in block {usage_key}")
+                continue
+
+            file_data = content_staging_api.get_staged_content_static_file_data(
+                self.staged_content.id,
+                original_filename,
+            )
+            if not file_data:
+                log.error(
+                    f"Staged content {self.staged_content.id} included referenced "
+                    f"file {original_filename}, but no file data was found."
+                )
+                continue
+
+            filename = f"static/{file_basename}"
+            media_type_str, _ = mimetypes.guess_type(filename)
+            if not media_type_str:
+                media_type_str = "application/octet-stream"
+
+            media_type = authoring_api.get_or_create_media_type(media_type_str)
+            content = authoring_api.get_or_create_file_content(
+                self.learning_package.id,
+                media_type.id,
+                data=file_data,
+                created=created_at,
+            )
+
+            try:
+                authoring_api.create_component_version_content(component_version.pk, content.id, key=filename)
+            except IntegrityError:
+                pass  # Content already exists
+
+
+def get_or_create_publishable_entity_mapping(usage_key, component) -> tuple[PublishableEntityMapping, bool]:
+    """
+    Creates a mapping between the source usage key and the target publishable entity.
+    """
+    if isinstance(component, Container):
+        target_package = component.publishable_entity.learning_package
+    else:
+        target_package = component.learning_package
+    return PublishableEntityMapping.objects.get_or_create(
+        source_usage_key=usage_key,
+        target_entity=component.publishable_entity,
+        target_package=target_package,
+    )
+
+
+def get_usage_key_string_from_staged_content(staged_content, block_id):
+    """
+    Get the usage ID from a staged content by block ID.
+    """
+    return next((block_usage_id for block_usage_id in staged_content.tags if block_usage_id.endswith(block_id)), None)
+
+
+def get_block_to_import(node, usage_key):
+    """
+    Get the block to import from a node.
+    """
+
+    if node.get('url_name') == usage_key.block_id:
```

Comment:
```suggestion
    if node.tag == usage_key.block_type and node.get('url_name') == usage_key.block_id:
```
A block_id (a.k.a. url_name) exists within the namespace of a block_type. For example, the same course could have `type@html+block@123` and `type@problem+block@123`. So, to properly identify a block in a course, check both the type and the ID.

Reply:
Good point, thanks.